### PR TITLE
[refactoring] xrdp: enable loading any keymap files

### PR DESCRIPTION
**Pull request by metalefty**
by separating functions. This makes it easier to create unit tests.

-------

## Background

I'm in the initial implementation stage of TOML config parser. Firstly, I started to TOML parser for keymap files.

I would like to confirm TOML parser is valid by comparing results of current INI parser and TOML parser are the same in unit tests.  For this purpose, I need functions that can load any keymap file.

Pseudo test code will be like this:

```C

struct xrdp_keymap *keymap_ini;
struct xrdp_keymap *keymap_toml;

get_keymaps_by_file("./km-00000409.ini", keymap_ini);
get_keymaps_by_file_toml("./km-00000409.toml", keymap_toml);

compare_xrdp_keymap(keymap_ini, keymap_toml);
```

### CODE REVIEW

**Comment by metalefty** [comment]
Changed function name a bit.

File: xrdp/lang.c [hunk 1]
```diff
@@ -258,36 +257,21 @@ get_keymaps(int keylayout, struct xrdp_keymap *keymap)
...
-            km_read_section(fd, "shiftcapslock", keymap->keys_shiftcapslock);
-            km_read_section(fd, "shiftcapslockaltgr", keymap->keys_shiftcapslockaltgr);
-
-            if (g_memcmp(lkeymap, keymap, sizeof(struct xrdp_keymap)) != 0)
-            {
-                LOG(LOG_LEVEL_WARNING,
-                    "local keymap file for 0x%08x found and doesn't match "
-                    "built in keymap, using local keymap file", keylayout);
-            }
+        lkeymap = (struct xrdp_keymap *)g_malloc(sizeof(struct xrdp_keymap), 0);
```
Comment (matt335672) [inline xrdp/lang.c:261]: Should probably add a check for malloc failure here.

File: xrdp/lang.c [hunk 2]
```diff
@@ -297,3 +281,33 @@ get_keymaps(int keylayout, struct xrdp_keymap *keymap)
...
+
+/*****************************************************************************/
+int km_load_file(const char *filename, struct xrdp_keymap *keymap)
+{
+    int fd;
+
+    fd = g_file_open(filename);
+    LOG(LOG_LEVEL_INFO, "Loading keymap file %s", filename);
+
+    if (fd != -1)
```
Comment (matt335672) [inline xrdp/lang.c:293]: Need to log error if keymap couldn't be opened

File: xrdp/xrdp.h [hunk 3]
```diff
@@ -411,6 +411,9 @@ get_char_from_scan_code(int device_flags, int scan_code, int *keys,
 int
 get_keymaps(int keylayout, struct xrdp_keymap *keymap);
 
+int
+km_load_file(const char *filename, struct xrdp_keymap *keymap);
+
```
Comment (matt335672) [inline xrdp/xrdp.h:416]: I notice this is a global. Is the intention to use it elsewhere later on?

**Review by matt335672 — COMMENTED** [review]
A few comments for you.

File: xrdp/lang.c (hunk 1 above)
Comment (metalefty) [reply xrdp/lang.c:261]: Agreed but this is the same as before. I would like to add check later.

File: xrdp/lang.c (hunk 2 above)
Comment (metalefty) [reply xrdp/lang.c:293]: Indeed.

File: xrdp/xrdp.h (hunk 3 above)
Comment (metalefty) [reply xrdp/xrdp.h:416]: Yes, this is intentional and I'm going to add some unit tests later.

**Review by matt335672 — APPROVED** [review]

**Pull request merged**